### PR TITLE
Add Notion

### LINKED ISSUE #896: Add integration with Notion
Feature request to add support for tracking time on [Notion](https://www.notion.so/).
Notion is a great Knowledge Base software. I'm starting to use it for documentation in most of my projects. And it's user base is growing rapidly.

### PR DESCRIPTION
**Pull request by karlhorky**
Hi there, thanks for this project, it is very useful!

Not sure if I'm doing this the right way, mostly just copied from the Trello implementation.

Visually it appears like this at the top of task cards:

screen shot 2018-12-03 at 15 50 58

### TODO before merging

- [x] [Remove `if`](https://github.com/toggl/toggl-button/pull/1203#discussion_r239537801)
- [x] [Move CSS to `style.css`](https://github.com/toggl/toggl-button/pull/1203#discussion_r239539151)
- [x] [Add `20px` `margin-right`](https://github.com/toggl/toggl-button/pull/1203#pullrequestreview-182349714)
- [x] [Simplify getting description text](https://github.com/toggl/toggl-button/pull/1203#discussion_r239538492)
- [x] [Minimal version with `buttonType: 'minimal'`](https://github.com/toggl/toggl-button/pull/1203#discussion_r239552882)

### Failed

- ~~[Simplify selector](https://github.com/toggl/toggl-button/pull/1203#discussion_r239537496)~~

### Further TODOs from @lewcart (https://github.com/toggl/toggl-button/issues/896#issuecomment-444808953)

- [ ] [Minimal button](https://github.com/toggl/toggl-button/blob/a5c7b8144098788e0bd99a7cae47baab54597159/src/scripts/content/asana.js#L187) for (unchecked) todos (like [Basecamp](https://github.com/toggl/toggl-button/blob/a5c7b8144098788e0bd99a7cae47baab54597159/src/scripts/content/basecamp.js#L74-L96))
- [ ] [Minimal button](https://github.com/toggl/toggl-button/blob/a5c7b8144098788e0bd99a7cae47baab54597159/src/scripts/content/asana.js#L187) for (incomplete) tasks in list view (like [Basecamp](https://github.com/toggl/toggl-button/blob/a5c7b8144098788e0bd99a7cae47baab54597159/src/scripts/content/basecamp.js#L74-L96))
- [ ] Add column [minimal button](https://github.com/toggl/toggl-button/blob/a5c7b8144098788e0bd99a7cae47baab54597159/src/scripts/content/asana.js#L187) for (incomplete) tasks in table view - how??
- [ ] Use the grayscale logo (may need to change [style.css](https://github.com/toggl/toggl-button/blob/3954ad26fe4ec948cbad9361aa1192afbe3d8ed2/src/styles/style.css) - [maybe difficult?](https://github.com/toggl/toggl-button/pull/1203#pullrequestreview-182349714))

Related to #896

### CODE REVIEW

**Comment by karlhorky** [comment]
@tcrammond I've added some todos from @lewcart (ref: https://github.com/toggl/toggl-button/issues/896#issuecomment-444808953). I would do at least the first todo (marked **TODO before merging**) before merging this.

The rest of them are effectively extra features (nice-to-haves) and can be done in a further pull request.

File: src/scripts/content/notion.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+togglbutton.render(
+  '.notion-overlay-container .notion-peek-renderer .notion-page-controls + .notion-selectable > [contenteditable="true"][placeholder="Untitled"]:not(.toggl)',
```
[the remark this answers is not in the document]
Comment (karlhorky) [reply src/scripts/content/notion.js]: Already tried something similar but it didn't work unfortunately. I just tried your suggestion and it also didn't work. Reverting to my version.

File: src/scripts/content/notion.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      descriptionElem = elem,
+      projectElem = $(
+        '.notion-sidebar-container > * > * > * > * > * > * > * > * + * > * > *'
+      ),
+      togglButtonLoc = $(
+        '.notion-overlay-container .notion-peek-renderer [rel="noopener noreferrer nofollow"] + [style*="flex-grow: 1"] + *'
+      );
+
+    if (!descriptionElem || !togglButtonLoc) {
+      return;
```
[the remark this answers is not in the document]
Comment (karlhorky) [reply src/scripts/content/notion.js]: Done

File: src/scripts/content/notion.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    descFunc = function() {
+      return descriptionElem.textContent;
```
[the remark this answers is not in the document]
Comment (karlhorky) [reply src/scripts/content/notion.js]: Done

File: src/scripts/content/notion.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    };
+
+    link = togglbutton.createTimerLink({
+      className: 'notion',
+      description: descFunc,
+      projectName: projectElem && projectElem.textContent,
+      calculateTotal: true
+    });
+
+    link.style.cursor = 'pointer';
```
[the remark this answers is not in the document]
Comment (karlhorky) [reply src/scripts/content/notion.js]: Done

File: src/scripts/content/notion.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    descFunc = function() {
+      return descriptionElem.textContent;
+    };
+
+    link = togglbutton.createTimerLink({
+      className: 'notion',
+      description: descFunc,
+      projectName: projectElem && projectElem.textContent,
+      calculateTotal: true
```
[the remark this answers is not in the document]
Comment (karlhorky) [reply src/scripts/content/notion.js]: Done

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "toggl-button",
-  "version": "1.14.0",
+  "version": "1.14.1",
```
[the remark this answers is not in the document]
Comment (karlhorky) [reply package.json]: Done

**Comment by karlhorky** [comment]
@tcrammond done :)

**Comment by tcrammond** [comment]
🙌 this will hopefully be released later this week. It may take some time from there to roll out to 100% of users.

**Comment by karlhorky** [comment]
Hey @xmarkclx, continuing the discussion here to avoid spamming everyone in the other thread.

Your image from https://github.com/toggl/toggl-button/issues/896#issuecomment-499448543 makes it appear like something has broken this feature 😕:

**Comment by kailovel** [comment]
@karlhorky is this still working with notion?

**Comment by alexjpanagis** [comment]
@karlhorky Any news? I'm also not seeing the button in Notion.

**Comment by karlhorky** [comment]
For anyone looking for a fix, this may be coming soon: https://github.com/toggl/toggl-button/pull/1492

**Pull request merged**